### PR TITLE
[MGDOBR-11] "Send to Kafka Topic" Action - Manager Side Implementation

### PR DESCRIPTION
**Pull request by robpblake**
This PR adds support on the manager side for a single Action on a Processor. More specifically in our case for this iteration, it adds only for a single, specific action:

- Send an event to a Kafka Topic

Action payload to the API is as follows (as part of the ProcessorRequest):

```json
"action": {
  "name": "my descriptive name",
  "type": "type-of-action",
  "parameters": {
      "action_specific_key": "action_specific_value"
  }
}
```

In addition, this PR adds the following:

- Processors must now be submitted with an Action. A processor without an Action is not valid
- The `parameters` for the "send to Kafka topic" action are validated to contain any string for the `topic` parameter

The assumptions made for this implementation:

- Our deployment is pre-configured with the Kafka Cluster that the Action topic will exist on (including security access)
- The Topic must already exist. Not attempt to create the topic will be made.

Shard side of the implementation will follow in another PR.

### CODE REVIEW

**Review by r00ta — APPROVED** [review]
For the Filters we have created a `BaseFilter` and multiple classes that extends it. In this case we only have one class and we put the action specific parameters in the generic `parameters` Map (in addition of the manual handling of the validation with `KafkaTopicActionParamsValidator` and similar). 

I think it would be nice to keep consistency in the codebase between the Filters and the Actions. 
Anyway, if you agree with that but prefer to move forward with the current implementation so to finalize the remaining tickets for the sprint, feel free to create a ticket that we will pick up later. 

Just a minor additional comment below

File: manager/src/main/java/com/redhat/service/bridge/manager/api/models/responses/ProcessorResponse.java [hunk 1]
```diff
@@ -41,6 +42,9 @@
     @JsonProperty("filters")
     private Set<BaseFilter> filters;
 
+    @JsonProperty("action")
+    private ActionRequest action;
```
[the remark this answers is not in the document]
Comment (robpblake) [reply manager/src/main/java/com/redhat/service/bridge/manager/api/models/responses/ProcessorResponse.java:46]: Yeah, I wasn't sure about the name. I'll go for `BaseAction` instead 👍

File: infra/src/main/java/com/redhat/service/bridge/infra/models/actions/ActionRequest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Map;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ActionRequest {
```
[the remark this answers is not in the document]
Comment (robpblake) [reply infra/src/main/java/com/redhat/service/bridge/infra/models/actions/ActionRequest.java:13]: Yep. I was close to using this anyways, so I'll rename.

**Comment by robpblake** [comment]
> For the Filters we have created a `BaseFilter` and multiple classes that extends it. In this case we only have one class and we put the action specific parameters in the generic `parameters` Map (in addition of the manual handling of the validation with `KafkaTopicActionParamsValidator` and similar).
> 
> I think it would be nice to keep consistency in the codebase between the Filters and the Actions.
> Anyway, if you agree with that but prefer to move forward with the current implementation so to finalize the remaining tickets for the sprint, feel free to create a ticket that we will pick up later.
> 
> Just a minor additional comment below

@r00ta I considered this, but my current thoughts are that we don't really need it for the Actions. They always follow the pattern of:

- name (which is a string)
- type (which is a string)
- parameters (which is a map of strings)

Whereas with `Filters`, we have a slightly different payload structure depending on the Filter you're using, which is a +1 for the approach you've taken. I'm not sure what the subclasses bring us on the user-facing API when it comes to Actions. 

I was going to see how the shard side implementation went. At some point we're gonna need to map the action `type` onto something that can do the actual wiring e.g make sure a Kafka topic exists and send the message to the topic, but again the `type` and generic `Map` of params might be OK there.

There were other things I thought about as well when doing the implementation:

- Action input schema could be defined in a schema registry, which would let us have a single approach to validation (remove the Kafka action specific stuff that we have here)
- I think longer-term we will need to be in the position to extend the platform with additional Actions, ideally without touching the API

Let me work-through the shard side in the next PR and I'll keep this in mind.

**Review by danielezonca — APPROVED** [review]

**Pull request merged**